pattern/constraint/thing/ValueConstraint.java

```diff
@@ -326,6 +363,22 @@ public java.lang.String toString() {
         public String clone(Conjunction.ConstraintCloner cloner) {
             return cloner.cloneVariable(owner).valueString(predicate(), value);
         }
+
+        @Override
+        public boolean isCompatibleWithEQ(ValueConstraint<?> conclusionValueConstraint) {
+            if (conclusionValueConstraint.isVariable()) return true;
+            if (!conclusionValueConstraint.isString()) return false;
+            if (predicate.isEquality()) {
+                return PredicateOperator.Equality.of(predicate.asEquality()).apply(Predicate.compareStrings(conclusionValueConstraint.asString().value(), value));
+            } else if (predicate.isSubString()) {
+                PredicateOperator.SubString operator = PredicateOperator.SubString.of(predicate.asSubString());
+                if (operator == PredicateOperator.SubString.CONTAINS) {
+                    return PredicateOperator.SubString.CONTAINS.apply(conclusionValueConstraint.asString().value(), value);
+                } else if (operator == PredicateOperator.SubString.LIKE) {
+                    return PredicateOperator.SubString.LIKE.apply(conclusionValueConstraint.asString().value(), Pattern.compile(value));
```

Comment:
Doing Pattern.compile here may be non-standard, but the standard way will likely require more cascading changes.